pkg/csi/service/wcp/controller.go

```diff
@@ -513,6 +513,21 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 	// Attach the volume to the node
 	diskUUID, err := common.AttachVolumeUtil(ctx, c.manager, podVM, req.VolumeId)
 	if err != nil {
+		if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FakeAttach) {
+			log.Infof("Volume attachment failed. Checking if it can be fake attached")
+			if canFakeAttach(ctx, req.VolumeId, c.manager.VolumeManager) {
+				publishInfo := make(map[string]string)
+				publishInfo[common.AttributeDiskType] = common.DiskTypeBlockVolume
+				publishInfo[common.AttributeFakeAttached] = "true"
+
+				resp := &csi.ControllerPublishVolumeResponse{
+					PublishContext: publishInfo,
+				}
+				log.Infof("Volume %s has been fake attached", req.VolumeId)
+				return resp, nil
+			}
+			log.Infof("Volume %s is not eligible to be fake attached", req.VolumeId)
```

Comment:
This log could be misleading if `canFakeAttach` failed with an error internally.

Reply:
Fixed.